Documentation/writing-principles-and-guidelines/spelling-and-typo3-terminology.rst

```diff
@@ -1,111 +1,1005 @@
+    -   *sys_log / sys_history*
 
-.. _4-e:
+        The :sql:`sys_log` and :sql:`sys_history` are database tables in TYPO3
+        for recording and tracking system events and changes.
 
-E
-^
+    ..  _sysext:
+    -   *sysext*
 
-Extbase: One word, small b.
+        Sysexts are SYStem EXTensions in TYPO3 that provide core functions
+        and features. This is the name of a central TYPO3 Core Sourcecode
+        directory, and in older TYPO3 versions there was a stronger separation
+        of system extension and local extension.
 
-.. _5-f:
+*   T
 
-F
-^
+    ..  _t3dd:
+    -   *T3DD*
 
-FlexForm,
+        T3DD stands for TYPO3 Developer Days, an annual conference for TYPO3
+        developers and enthusiasts.
 
-Fluid: The Fluid Templating Engine.
+    ..  _tca:
+    -   *TCA*
 
-FormEngine
+        The Table Configuration Array (TCA) is a central configuration structure
+        in TYPO3 for defining and managing database tables and fields.
 
-Frontend: one word.
+    ..  _tceforms:
+    -   *TCEforms*
 
-.. _6-i:
+        TCEforms are forms in TYPO3 used for editing database entries and
+        content elements in the backend.
 
-I
-^
+    ..  _templates---fluid:
+    -   *Templates (=Fluid)*
 
-Introduction Package: Title case.
+        Templates in TYPO3, often created with Fluid, define the structure and
+        layout of the frontend display.
 
-.. _7-l:
+    ..  _ter:
+    -   *TER*
 
-L
-^
+        The TYPO3 Extension Repository (TER) is a central directory for
+        distributing TYPO3 extensions.
 
-LinkHandler: One word, camel case.
+    ..  _testing-framework:
+    -   *Testing Framework*
 
-.. _8-p:
+        The Testing Framework in TYPO3 provides tools and methods for conducting
+        automated tests used for developing the TYPO3 Core or custom projects
+        and extensions.
 
-P
-^
+    ..  _tsconfig:
+    -   *TSconfig*
 
-page TSconfig
+        TSconfig uses the TypoScript configuration language in TYPO3 and is used
+        for defining backend settings and configuration options. This is
+        seperated into "User TSConfig" and "Page TSConfig"
 
-.. _9-r:
+*   U
 
-R
-^
+    ..  _uid:
+    -   *uid*
 
-reStructuredText: Abbreviated as ‘reST’.
+        :sql:`uid` stands for Unique Identifier and is a unique identifier for
+        records and objects in the TYPO3 system. It complements the :sql:`pid`
+        (parent identifier).
 
-RTE: rich text editor.
+    ..  _upgrade-wizard:
+    -   *upgrade wizard*
 
-.. _10-s:
+        The upgrade wizard is a module in the TYPO3 backend used for performing
+        system and database upgrades.
 
-S
-^
+*   V
 
-sitepackage (one word, all lower case)
+    ..  _vendor:
+    -   *Vendor*
 
-system extension (not ‘sys extension’)
+        The term "Vendor" is most commonly used as a semantic grouping
+        identifier for PHP namespaces in developed extensions. Composer collects
+        all packages in a directory, that is also usually called "vendor" and
```

Comment:
```suggestion
        packages in a directory, that is also usually called "vendor" and
```

Reply:
IMO "All" is important here for developers, since you cannot have any packages outside of it. So it really is a full "all".